### PR TITLE
Teacher Development SMG Reporting

### PR DESCRIPTION
**Pull request by cristinabaldor**
# Pull Request

## Summary & Motivation

[//]: # "When merged, this pull request will..."

## Self-review

- [ ] Update **due date**, **assignee**, and **priority** on our
      [TEAMster Asana Project](https://app.asana.com/0/1205971774138578/1205971926225838)

  _If this is a same-day request, please flag that in Slack_

- [ ] <kbd>Format</kbd> has been run on all modified files

- [ ] Ensure you are using the `union_dataset_join_clause()` macro for queries that employ any
      models from these datasets:
  - deanslist
  - edplan
  - iready
  - pearson
  - powerschool
  - renlearn
  - titan

## Troubleshooting

- [SqlFluff Rules Reference](https://docs.sqlfluff.com/en/stable/rules.html)
- [Trunk](https://teamschools.github.io/teamster/CONTRIBUTING/#trunk)
- [dbt](https://teamschools.github.io/teamster/CONTRIBUTING/#dbt-cloud_1)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208024019845421

### CODE REVIEW

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__teacher_development.sql [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    coalesce(sr.preferred_name_lastfirst, td.observer_name) as observer_name,
+    if(sr.department_home_name = 'Teacher Development', 'TDT', 'NTNC') as observer_team,
+
+    os.final_score as performance_management_final_score,
+    os.final_tier as performance_management_final_tier,
+from {{ ref("int_performance_management__teacher_development") }} as td
+left join
+    {{ ref("base_people__staff_roster_history") }} as srh
+    on td.employee_number = srh.employee_number
+    and cast(td.observed_at as timestamp)
```
Comment (cristinabaldor) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__teacher_development.sql:42]: hey @cbini I'm trying to use the observed_at_timestamp field upstream in int_performance_management__teacher_development but when I run it without casting to timestamp I get an error on this join

File: src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__teacher_development.sql [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            o.observation_id,
+            o.rubric_name,
+            o.observation_score,
+            o.observed_at_timestamp,
+            o.academic_year,
+            o.observation_type,
+            o.observation_type_abbreviation,
+            o.observation_course,
+            o.observation_grade
+    ),
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__teacher_development.sql:38]: this stuff should be moved downstream to the tableau view

**Review by cbini — CHANGES_REQUESTED** [review]

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__teacher_development.sql (hunk 1 above)
Comment (cbini) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__teacher_development.sql]: that column is a date on the source table, needed to be converted to timestamp anyway

**Comment by cristinabaldor** [comment]
hey @bini, i can't get the timestamp change to build because `stg_adp_workforce_now__workers` is failing, but i think it works

**Review by cbini — APPROVED** [review]

**Pull request merged**